### PR TITLE
Remove linear regression normalization

### PR DESCRIPTION
**Pull request by YuhanLiin**
Remove normalization logic in `linfa-linear`, since that's handled in `linfa-preprocessing`.

Rewrite intercept fitting calculations.

### CODE REVIEW

File: algorithms/linfa-linear/src/ols.rs [hunk 1]
```diff
@@ -267,7 +190,7 @@ mod tests {
     /// fit we only need to provide one point.
     #[test]
     fn without_intercept_fits_line_through_origin() {
-        let lin_reg = LinearRegression::new().with_intercept(false);
+        let lin_reg = LinearRegression::new().fit_intercept(false);
```
Comment (bytesnake) [inline algorithms/linfa-linear/src/ols.rs:193]: can you keep the original name here, writing `LinearRegression::new().fit_intercept(false).fit(..)` sounds weird

**Review by bytesnake — APPROVED** [review]
:+1: let's get rid of the `ndarray_stats` dependency here

File: algorithms/linfa-linear/src/ols.rs (hunk 1 above)
Comment (YuhanLiin) [reply algorithms/linfa-linear/src/ols.rs:193]: Should I change it in logistic regression as well?

Comment (bytesnake) [reply algorithms/linfa-linear/src/ols.rs:193]: yes to keep the naming consistent

Comment (YuhanLiin) [reply algorithms/linfa-linear/src/ols.rs]: Turns out the name in logistic was already `with_intercept`

**Pull request merged**